python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -799,10 +799,42 @@ def nth(self, n):
         """
         Return the nth row from each group.
         """
-        result = self.agg(lambda x: x.nth(n)).sort_index()
+
+        new_df = self.obj.copy(deep=False)
+        new_df["__groupbynth_order__"] = range(0, len(self.obj))
+        gb = new_df.groupby(
+            by=self._by,
+            level=self._level,
+            sort=self._sort,
+            as_index=self._as_index,
+            dropna=self._dropna,
+            group_keys=self._group_keys,
+        )
+        result = gb.agg(lambda x: x.nth(n))
+        grouping_cols = (
+            self._by
+            if isinstance(self._by, tuple)
+            else tuple(
+                self._by,
+            )
+        )
+        col_idxs = [
+            i for i, x in enumerate(self.obj._data.names) if x in grouping_cols
+        ]
+        result = result.sort_index()
         sizes = self.size().sort_index()
 
-        return result[sizes > n]
+        result = result[sizes > n]
```

Comment:
Would it be more performant to do this filter then call `sort_index`?

Reply:
`sort_index` was being done to have a matching comparison in `sizes>n`, I removed the sorting of index twice and reduced it to reindexing only once.